converter/dicom-cast/src/Microsoft.Health.DicomCast.TableStorage/Features/Storage/TableServiceClientInitializer.cs

```diff
@@ -28,32 +29,29 @@ public class TableClientInitializer : ITableClientInitializer
         }
 
         /// <inheritdoc />
-        public CloudTableClient CreateTableClient()
+        public TableServiceClient CreateTableServiceClient()
         {
-            _logger.LogInformation("Creating TableClient instance");
-
-            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(_configuration.ConnectionString);
-
-            // TODO: Add retry policy for acessing the table storage
-            var tableRequestOptions = new TableRequestOptions();
-
-            CloudTableClient tableClient = storageAccount.CreateCloudTableClient(new TableClientConfiguration());
+            // TODO: Add retry policy for accessing the table storage
+            if (!string.IsNullOrWhiteSpace(_configuration.ConnectionString))
+            {
+                return new TableServiceClient(_configuration.ConnectionString);
+            }
 
-            return tableClient;
+            var credential = new DefaultAzureCredential(new DefaultAzureCredentialOptions { ManagedIdentityClientId = _configuration.ManagedIdentityClientId });
```

Comment:
You can also let the Azure APIs handle this for you through the use of the package [`Microsoft.Extensions.Azure`](https://www.nuget.org/packages/Microsoft.Extensions.Azure/). They have a type `AzureComponentFactory` that has a method `CreateTokenCredential`. You give it the `IConfigurationSection` for the credentials and it creates the correct `TokenCredential` option (or the `DefaultAzureCredential` if none are specified). However, if we need logic like "use this managed identity if found, otherwise use defaults," then we may not be able to use it.

The configuration section looks for names like `Credential` = `"managedidentity"` and `ClientId`. I had to look up these values in their source code [here](https://github.com/Azure/azure-sdk-for-net/blob/e1ec26c5401e836fe3ddee778423b1a628949599/sdk/extensions/Microsoft.Extensions.Azure/src/Internal/ClientFactory.cs#L86).

Reply:
If you went this route, you could have `Connection` either be a connection string or a section containing more configuration info about the connection.